ovh/resource_cloud_project_network_private_test.go

```diff
@@ -2,158 +2,49 @@ package ovh
 
 import (
 	"fmt"
-	"log"
 	"os"
-	"strings"
 	"testing"
-	"time"
 
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 )
 
-var testAccCloudProjectNetworkPrivateConfig_attachVrack = `
-resource "ovh_vrack_cloudproject" "attach" {
-  service_name = "%s"
-  project_id   = "%s"
-}
-
-data "ovh_cloud_project_regions" "regions" {
-  service_name = ovh_vrack_cloudproject.attach.project_id
-
-  has_services_up = ["network"]
-}
-`
-
-var testAccCloudProjectNetworkPrivateConfig_noAttachVrack = `
-data "ovh_cloud_project_regions" "regions" {
-  service_name = "%s"
-
-  has_services_up = ["network"]
-}
-`
-
-var testAccCloudProjectNetworkPrivateConfig_basic = `
-%s
-
-resource "ovh_cloud_project_network_private" "network" {
-  service_name = data.ovh_cloud_project_regions.regions.service_name
-  vlan_id    = 0
-  name       = "terraform_testacc_private_net"
-  regions    = slice(sort(tolist(data.ovh_cloud_project_regions.regions.names)), 0, 3)
-}
-`
-
-func testAccCloudProjectNetworkPrivateConfig(config string) string {
-	attachVrack := fmt.Sprintf(
-		testAccCloudProjectNetworkPrivateConfig_attachVrack,
-		os.Getenv("OVH_VRACK_SERVICE_TEST"),
-		os.Getenv("OVH_CLOUD_PROJECT_SERVICE_TEST"),
-	)
-	noAttachVrack := fmt.Sprintf(
-		testAccCloudProjectNetworkPrivateConfig_noAttachVrack,
-		os.Getenv("OVH_CLOUD_PROJECT_SERVICE_TEST"),
-	)
-
-	if os.Getenv("OVH_ATTACH_VRACK") == "0" {
-		return fmt.Sprintf(
-			config,
-			noAttachVrack,
-		)
-	}
-
-	return fmt.Sprintf(
-		config,
-		attachVrack,
-	)
-}
-
-func init() {
-	resource.AddTestSweepers("ovh_cloud_project_network_private", &resource.Sweeper{
-		Name: "ovh_cloud_project_network_private",
-		F:    testSweepCloudProjectNetworkPrivate,
-	})
-}
-
-func testSweepCloudProjectNetworkPrivate(region string) error {
-	client, err := sharedClientForRegion(region)
-	if err != nil {
-		return fmt.Errorf("error getting client: %s", err)
-	}
-
-	vrack := os.Getenv("OVH_VRACK_SERVICE_TEST")
-	if vrack == "" {
-		log.Print("[DEBUG] OVH_VRACK_SERVICE_TEST is not set. No cloud_network_private to sweep")
-		return nil
-	}
-
-	projectId := os.Getenv("OVH_CLOUD_PROJECT_SERVICE_TEST")
-	if projectId == "" {
-		log.Print("[DEBUG] OVH_CLOUD_PROJECT_SERVICE_TEST is not set. No cloud_network_private to sweep")
-		return nil
-	}
-
-	networks := []CloudProjectNetworkPrivateResponse{}
-	err = client.Get(fmt.Sprintf("/cloud/project/%s/network/private", projectId), &networks)
-	if err != nil {
-		return fmt.Errorf("error listing private networks for project %q:\n\t %q", projectId, err)
+var testCreateNetworkPrivate = fmt.Sprintf(`
```

Comment:
could you declare these configs in the test func directly ? They don't need to be global